test/shopify-cli/git_test.rb

```diff
@@ -0,0 +1,156 @@
+require 'test_helper'
+
+module ShopifyCli
+  class GitTest < MiniTest::Test    
+    def setup
+      super 
+
+      @status_mock = {
+        false: mock,
+        true: mock,
+      }
+      @status_mock[:false].stubs(:success?).returns(false)
+      @status_mock[:true].stubs(:success?).returns(true)
+    end
+
+    def test_branches_returns_master_if_no_branches_exist
+      @context.expects(:capture2e)
+        .with('git', 'branch', '--list', '--format=%(refname:short)')
+        .returns(['', @status_mock[:true]])
+
+      git_service = ShopifyCli::Git.new(@context)
+
+      assert_equal(['master'], git_service.branches)
+    end
+
+    def test_branches_returns_list_of_branches_if_multiple_exist
+      @context.expects(:capture2e)
+        .with('git', 'branch', '--list', '--format=%(refname:short)')
+        .returns(["master\nsecond_branch\n", @status_mock[:true]])
+
+      git_service = ShopifyCli::Git.new(@context)
+
+      assert_equal(['master', 'second_branch'], git_service.branches)
+    end
+
+    def test_branches_raises_if_finding_branches_fails
+      @context.stubs(:capture2e)
+        .with('git', 'branch', '--list', '--format=%(refname:short)')
+        .returns(['', @status_mock[:false]])
+      git_service = ShopifyCli::Git.new(@context)
+
+      assert_raises ShopifyCli::Abort do
+        git_service.branches
+      end
+    end
+
+    def test_init_initializes_successfully
+      stub_git_init(status: true, commits: true)
+      git_service = ShopifyCli::Git.new(@context)
+
+      capture_io do
+        git_service.init
+      end
```

Comment:
Not entirely sure what `capture_io` is doing or if this is what I want to use, but other tests were using it 🤷‍♀
This method will return `nil` if it is successful.

Reply:
capture_io is basically just keeping an array of the output to the console so that you can compare the output to see if it logged the appropriate messages

Reply:
So maybe not the best use of it here. I think I'll switch to `assert_nil`.